### PR TITLE
feat(serving): return estimates for all ref edge subkinds

### PR DESCRIPTION
**Pull request by salguarnieri**
Instead of returning a total count of ref edges, return a map containing the count of each ref subkind. With ref/writes, clients want to have more visibility into the counts of different ref kinds.

### CODE REVIEW

File: kythe/go/serving/xrefs/xrefs_test.go [hunk 1]
```diff
@@ -1330,11 +1330,18 @@ func TestCrossReferences(t *testing.T) {
 
 	if err := testutil.DeepEqual(&xpb.CrossReferencesReply_Total{
 		Definitions: 1,
-		References:  2,
+		RefEdgeToCount: map[string]int64{
+			"%/kythe/edge/ref": 2,
```
Comment (schroederc) [inline kythe/go/serving/xrefs/xrefs_test.go:1334]: Get rid of the leading `%` in the counts

File: kythe/go/serving/xrefs/xrefs.go [hunk 2]
```diff
@@ -758,8 +758,11 @@ func (t *Table) CrossReferences(ctx context.Context, req *xpb.CrossReferencesReq
 		CrossReferences: make(map[string]*xpb.CrossReferencesReply_CrossReferenceSet, len(req.Ticket)),
 		Nodes:           make(map[string]*cpb.NodeInfo, len(req.Ticket)),
 
-		Total: &xpb.CrossReferencesReply_Total{},
+		Total: &xpb.CrossReferencesReply_Total{
+			RefEdgeToCount: make(map[string]int64),
```
Comment (schroederc) [inline kythe/go/serving/xrefs/xrefs.go:762]: Add these together for the existing References field until it is removed.

**Review by schroederc — CHANGES_REQUESTED** [review]

File: kythe/go/serving/xrefs/xrefs_test.go (hunk 1 above)
Comment (salguarnieri) [reply kythe/go/serving/xrefs/xrefs_test.go]: done

File: kythe/go/serving/xrefs/xrefs.go (hunk 2 above)
Comment (salguarnieri) [reply kythe/go/serving/xrefs/xrefs.go:764]: I added the old code back in until the field can be deleted

File: kythe/go/serving/xrefs/xrefs_test.go [hunk 3]
```diff
@@ -1331,10 +1331,18 @@ func TestCrossReferences(t *testing.T) {
...
+		RefEdgeToCount: map[string]int64{
+			"/kythe/edge/ref": 2,
+		},
 	}, reply.Total); err != nil {
 		t.Error(err)
 	}
-	if err := testutil.DeepEqual(&xpb.CrossReferencesReply_Total{}, reply.Filtered); err != nil {
+	wantFiltered := &xpb.CrossReferencesReply_Total{
+		RefEdgeToCount: map[string]int64{
+			"/kythe/edge/ref": 0,
```
Comment (schroederc) [inline kythe/go/serving/xrefs/xrefs_test.go:1342]: Do we want these superfluous zero counts?

Comment (salguarnieri) [reply kythe/go/serving/xrefs/xrefs_test.go]: I added code to remove them.

**Pull request merged**